### PR TITLE
Add remove and remote_url options

### PR DESCRIPTION
**Pull request by neumayr**
Ciao @michelegera

Needed these options for a project of mine.

* add `remove` option
  * https://github.com/carrierwaveuploader/carrierwave#removing-uploaded-files
* add `remote_url` option
  * https://github.com/carrierwaveuploader/carrierwave#uploading-files-from-a-remote-location
* add `text_field` for remote_FIELD_url
* add `check_box` for remove_FIELD

### CODE REVIEW

**Comment by michelegera** [comment]
Ciao @neumayr, thanks for this PR! Going to take a look at it today and let you know. Thanks for your patience 😄

File: app/views/fields/carrierwave/_form.html.erb [hunk 1]
```diff
@@ -18,9 +18,11 @@ attachment(s) for this attribute
 </div>
 <div class="field-unit__field">
   <%= f.file_field field.attribute, multiple: field.multiple? %>
+  <%= f.text_field "remote_#{field.attribute}_url", placeholder: "remote url" if field.remote_url? %>
```
Comment (michelegera) [inline app/views/fields/carrierwave/_form.html.erb]: For style consistency, can you please change the placeholder option to `placeholder: 'Remote URL'`?

File: app/views/fields/carrierwave/_form.html.erb [hunk 2]
```diff
@@ -18,9 +18,11 @@ attachment(s) for this attribute
...
   <br />
   <%= render 'fields/carrierwave/show', field: field %>
 
-  <%= f.text_field "remote_#{field.attribute}_url" if field.remote_url? %>
-  <%= f.check_box "remove_#{field.attribute}" if field.remove? %>
+  <% if field.remove? %>
+    <label><%= f.check_box "remove_#{field.attribute}" %> <%= "Remove #{field.attribute}" %></label>
```
Comment (michelegera) [inline app/views/fields/carrierwave/_form.html.erb]: Please use Rails’ form helper for the label too:
```
<%= f.label "remove_#{field.attribute}" do %>
  <%= f.check_box "remove_#{field.attribute}" %> Remove <%= field.attribute %>
<% end %>
```

**Comment by michelegera** [comment]
Hi @neumayr!

First of all, thanks for your work on this.

Can you please amend your commits and address the review notes I’ve left? It’s just a couple of minor stylistic changes.

After that, please squash all the four commits into one – the commit message should be capitalize and could be something along the lines of “Add ability to remove files and upload them from remote URLs“.

Thanks!

**Comment by neumayr** [comment]
hey @michelegera 

I'm on the go currently, for this time I managed to do all the changes and commits only using github's web UI. Nice how far they push it. This means I can't do the squash over the web UI. 
But there is a new setting to allow squashing for pull requests. Did you know? 
This way a maintainer could squash and set the final commit message. It's worth a try…

:octocat: https://help.github.com/articles/configuring-commit-squashing-for-pull-requests/

**Comment by michelegera** [comment]
@neumayr thanks for those changes on the go! I’ve squashed and merged your PR.

I will release a new version of the gem later today.

**Pull request merged**